### PR TITLE
chore: setup scheduled release

### PR DESCRIPTION
**Pull request by eunjae-lee**
**Summary**

This PR sets up scheduled release. And it additionally upgrades `ship.js` to the latest version and updated some configs due to the breaking changes from `ship.js`.

We can periodically release the newer version of the library. And I hope to get rid of unnecessary communication like:

> Can I release this today? Any blocker?

Instead, the pull request will be created automatically, and as reviewers, we can address any blocker in the pull request if there is any.

**Result**

Every Tuesday 09:00 UTC (11am Paris), `shipjs prepare` will run and create a release pull-request.
If there is a patch update and there is no `fix` commit, then it will skip.

### CODE REVIEW

File: ship.config.js [hunk 1]
```diff
@@ -20,10 +19,9 @@ module.exports = {
   beforeCommitChanges: ({ exec }) => {
     exec('yarn doctoc');
   },
-  pullRequestTeamReviewer: ['instantsearch-for-websites'],
+  pullRequestTeamReviewers: ['instantsearch-for-websites'],
   buildCommand: ({ version }) =>
     `NODE_ENV=production VERSION=${version} yarn build`,
-  testCommandBeforeRelease: () => 'echo "No need to test again."',
```
Comment (eunjae-lee) [inline ship.config.js:26]: The default behavior of the latest shipjs skips this test step, so we don't need this any more.

File: ship.config.js [hunk 2]
```diff
@@ -33,10 +31,9 @@ module.exports = {
     }
   },
   slack: {
-    // disable slack notification for `prepared` and `releaseStart` lifecycle.
+    // disable slack notification for `prepared` lifecycle.
```
Comment (eunjae-lee) [inline ship.config.js:34]: `releaseStart` lifecycle has been removed from shipjs.

File: ship.config.js [hunk 3]
```diff
@@ -20,10 +19,9 @@ module.exports = {
   beforeCommitChanges: ({ exec }) => {
     exec('yarn doctoc');
   },
-  pullRequestTeamReviewer: ['instantsearch-for-websites'],
+  pullRequestTeamReviewers: ['instantsearch-for-websites'],
```
Comment (eunjae-lee) [inline ship.config.js:22]: renamed in shipjs.

**Review by Haroenv — APPROVED** [review]

File: .circleci/config.yml [hunk 4]
```diff
@@ -135,3 +150,18 @@ workflows:
       - type check js (optional)
       - e2e tests
       - release if needed
+          filters:
```
Comment (Haroenv) [inline .circleci/config.yml:153]: this probably needs to be part of `triggers` or a level lower?

Comment (eunjae-lee) [reply .circleci/config.yml:153]: https://github.com/algolia/shipjs/blob/main/.circleci/config.yml#L90-L96

It works like that.
I hate that I need to find an example every time I modify circleci config :(

**Review by Haroenv — APPROVED** [review]
as far as I can tell this is good

**Pull request merged**